src/Projection/PdoEventStoreProjector.php

```diff
@@ -180,7 +187,8 @@ public function __construct(
         int $persistBlockSize,
         int $sleep,
         bool $triggerPcntlSignalDispatch = false,
-        int $updateLockThreshold = 0
+        int $updateLockThreshold = 0,
+        GapDetection $gapDetection = null
```

Comment:
I really like how the amount of constructor arguments grow and grows... way beyond anything initially designed. But that's not your fault here, just something I noticed.

Reply:
yeah, one reason why I've put the gap detection stuff into a class and use it as config option. otherwise we would have 3 more constructor args ;)